### PR TITLE
influxdb-cxx: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **influxdb-cxx/0.7.1**

Try to close https://github.com/conan-io/conan-center-index/issues/19125.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by steinerthomas — APPROVED** [review]

File: recipes/influxdb-cxx/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # BUILD_SHARED_LIBS is defined explicitly in CMakeLists.txt
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["INFLUXCXX_TESTING"] = False
+        tc.cache_variables["INFLUXCXX_WITH_BOOST"] = self.options.boost
+        if self.options.shared:
+            tc.preprocessor_definitions["InfluxDB_EXPORTS"] = 1
```
Comment (jcar87) [inline recipes/influxdb-cxx/all/conanfile.py:80]: This one looks a little unusual - is it not possible to build the shared variant without the consumers defining preprocessor definitions externally?

**Review by jcar87 — COMMENTED** [review]
Thanks @toge - recipe looks great!! Just one small question

File: recipes/influxdb-cxx/all/conanfile.py (hunk 1 above)
Comment (offa) [reply recipes/influxdb-cxx/all/conanfile.py:80]: This shouldn't be necessary as the define is handled by CMakes GenerateExportHeader. However, there are reports of issues on MSVC. I wasn't able to reproduce the issue, but I'm no MSVC dev.

**Review by danimtb — APPROVED** [review]

File: recipes/influxdb-cxx/all/conanfile.py (hunk 1 above)
Comment (jcar87) [reply recipes/influxdb-cxx/all/conanfile.py:81]: Thanks @toge  - there is indeed a pre-existing issue, I've updated the recipe to reference it

I've had a chance to investigate the root cause and made a comment here:
https://github.com/offa/influxdb-cxx/issues/194#issuecomment-1935661442

File: recipes/influxdb-cxx/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#include <InfluxDBFactory.h>
+#include <InfluxDBException.h>
+#include <iostream>
+
+int main()
+{
+    try {
+        auto influxdb = influxdb::InfluxDBFactory::Get("xyz://foobar");
```
Comment (jcar87) [inline recipes/influxdb-cxx/all/test_package/test_package.cpp:8]: I've changed this to _not_ reference localhost - in the odd chance something is actually running.  Test packages in Conan center _must not_ make network requests to any resource whatsover.

**Review by jcar87 — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**